regenerate-initrd-posttrans

```diff
@@ -45,12 +115,21 @@ fi
 
 for f in "$dir"/*; do
 	case $f in
-	    "$dir/*")
+		"$dir/*")
 		[ -e "$f" ] || break;;
 	esac
 	# check if we are in a build chroot
-	if ! [  -f /etc/fstab -a ! -e /.buildenv -a -x "$DRACUT" ] ; then
-		echo "Please run \"$DRACUT -f --regenerate-all\" as soon as your system is complete." >&2
+	if ! [  -f /etc/fstab -a ! -e /.buildenv ]; then
+		case "$initrd_generator" in
+			"dracut")
+				echo "Please run \"$DRACUT -f --regenerate-all\" as soon as your system is complete." >&2
+				;;
+			"mkosi-initrd")
+				# FIXME: mkosi-initrd does not provide anything like --regenerate-all yet
+				echo "Please regenerate all the initrds with \"$MKOSI_INITRD\" as soon as your system is complete." >&2
+				echo "WARNING: mkosi-initrd cannot generate the initrd directly into /boot, you need to use an intermediate directory." >&2
```

Comment:
Not sure if this warning is necessary; people using `mkosi-initrd` should know what they're doing.